pkg/controller/replication/switchover.go

```diff
@@ -46,8 +46,8 @@ func (r *ReplicationReconciler) reconcileSwitchover(ctx context.Context, req *re
 
 	phases := []switchoverPhase{
 		{
-			name:      "Set read_only in current primary",
```

Comment:
I think we should keep this phase as well, so we mark the current primary read-only and it will stay read-only after the failover as it will become a replica.

Reply:
What would be the benefit of setting it here? 

The current primary has already been locked in the previous step, so no new writes will happen in this instance. Also, as part or the primary and replica reconciliation we always set `read_only` idempotently, so I believe it should cover this case?


Primary:
https://github.com/mariadb-operator/mariadb-operator/blob/b02782032fcc9e459ef53ae2c51d28c014fd8468/pkg/controller/replication/config.go#L50

Replica:
https://github.com/mariadb-operator/mariadb-operator/blob/b02782032fcc9e459ef53ae2c51d28c014fd8468/pkg/controller/replication/config.go#L75

Maybe I am missing something?

Reply:
Well, my point was here that as you know when the connection is closed it unlocks tables automatically. As you already said it should be fine. But if somehow this connection is closed in the middle of the failover process for some reason like network etc. having the old primary as read_only would be safer than the writeable one

> In our case, the connections last one reconciliation cycle, so it should be fine

Reply:
> But if somehow this connection is closed in the middle of the failover process for some reason like network etc. having the old primary as read_only would be safer than the writeable one

Great, I think this is a valid point, I will add and extra step for this 👍🏻